frontend/src/stores/GraphDataStore.ts

```diff
@@ -0,0 +1,35 @@
+import { injectable } from 'inversify';
+import 'reflect-metadata';
+import { QueryResult } from '../shared/queries';
+import SimpleStore from './SimpleStore';
+import FilterStore from './FilterStore';
+import { FilterService } from '../services/filter';
+import { CancellationTokenSource } from '../utils/CancellationToken';
+
+@injectable()
+export default class GraphDataStore extends SimpleStore<QueryResult> {
```

Comment:
This name can be a bit misleading, since there is the graph data that is directly displayed as a `VisGraph`. This is a converted `QueryResult`. This `GraphDataStore` stores `QueryResult`s and not directly the graph data. So I would consider replacing the name to smth like `QueryResultStore` or indicate something like that as a doc. I would prefer `QueryResultStore`. So the pattern is `<T>Store`.